From d4852fcf561f6c9a3b7949011e419ad550dcb2e2 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 05 Aug 2021 10:32:07 +0300
Subject: [PATCH] [SW-50880] ppboot: greco_arc: move axi drain to early init

This is preparatory patch for running PCIe initialization code on
ARC0 for secureboot flow and allow ARC1 to continue with preboot
execution. In order to achieve that, the axi drain init which
is done after init of ARC0 in hl_plat_init needs to be moved to
early init function and should be done before ARC0 init. In later patch,
we will move to init of ARC0 to early init as well.

Change-Id: I62c69e0b776579296f47f62702b1a5eb22cf42ce
---

diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 2873fcb..86bab10 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -124,6 +124,9 @@
 	/* Functionality used to be done in BTL */
 	init_prstn();
 
+	/* PSOC AXI drain */
+	axi_enable_drain();
+
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_IATU_CONF_EN, true);
 
 	/*
@@ -161,8 +164,6 @@
 void hl_plat_init(void)
 {
 	init_arc();
-
-	axi_enable_drain();
 
 	sw_all_rst_init();
 }
